UnattendResources/Logon.ps1

```diff
@@ -190,8 +194,14 @@ try
     $purgeUpdates = Get-IniFileValue -Path $configIniPath -Section "updates" -Key "purge_updates" -Default $false -AsBoolean
     $disableSwap = Get-IniFileValue -Path $configIniPath -Section "sysprep" -Key "disable_swap" -Default $false -AsBoolean
     $goldImage = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "gold_image" -Default $false -AsBoolean
+    $goldImagePath = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "gold_image_path"
+    $productKey = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "product_key"
     $serialPortName = Get-IniFileValue -Path $configIniPath -Section "cloudbase_init" -Key "serial_logging_port"
 
+    if ($productKey -and $goldImagePath) {
+        Activate-Windows

```

Comment:
put a param to Activate-Windows -- don't use variables from the implicit context, as it is hard to read the code